### PR TITLE
Introduce "individual-item" type to domain picker items

### LINKED ISSUE #46862: Focused Launch: Domain step
If the site doesn't have a paid domain associated:
  - [x] Show the domain picker in the "active" group
  - [x] Step title is "Confirm your domain"
  - [x] Side commentary:
    - [x] On desktop, reads  "46.9% of globally registered domains are .com" and shows additional list items — PR #47281
    - [x] On mobile, starts with a bulb icon, reads "46.9% of registered domains are .com" and doesn't show any list items — PR #47281
  - [x] Show 3 domains (one is the default .wordpress.com)
  - [x] Show a "View all domains" link which changes the view to the domains picker details
  - [x] New "domain item" design - PR #47152  
  - [x] Show blue highlight for selected domain item - PR #47152 
  - [x] Pre-select the domain, by looking at the `@automattic/launch` data-store - PR #47191
  - [ ] If the `@automattic/launch` data-store doesn't contain a domain, fallback to items in the cart (if any) | #47233 
  - [ ] Ensure selected domain is always shown to the user (append it as the last paid domain if not in the initial list) #47540

Screenshot 2020-11-06 at 16 04 29

Screenshot 2020-10-28 at 15 13 42

Screenshot 2020-11-10 at 15 48 39

If the site already has a pre-purchased domain associated:
  - [x] show the domain in the "disabled" group
  - [x] Step title is 'Your domain'
  - [x] Show an info icon with a popover next to the title that reads "Changes to your purchased domain can be managed from your Domains page."
  - [x] Show only the purchased domain, which can't be selected by the user
  - [x] Side commentary:
    - [x] On desktop, reads  "Unique domains help build brand recognition and trust" — PR #47281
    - [x] On mobile, starts with a bulb icon, and reads "Unique domains help build brand trust" — PR #47281

Screenshot 2020-11-06 at 16 04 15

Screenshot 2020-11-10 at 15 48 09

### PR DESCRIPTION
**Pull request by alshakero**
### Changes proposed in this Pull Request

This PR introduces yet a new mode to the domain picker. Now we have 3 item types: `radio`, `button`, and `individual-item`. I'm not very happy with the term `individual`, I mean each element is on its own. 

The details on the changes are shown in the testing steps.

### Testing instructions - requires sandboxing because needs existing domains

1. In apps/editing-toolkit run `yarn dev --sync`.
2. Sandbox testingfocusedflow.wordpress.com.
3. Visit https://testingfocusedflow.wordpress.com/wp-admin/post-new.php.
4. In console, type `wp.data.dispatch('automattic/launch' ).openFocusedLaunch()`.

#### On Desktop

1. Selected item should be bordered with blue and should have blue text for domain name.
2. Hovered item should be **only** bordered by blue.
3. Simulate selecting the recommended domain by adding "is-selected" class to the item in DevTools.
4. The domain name, tld, "First year ...", price, and the badge, all should be blue. Price should have 0.68 opacity.
5. "First year ..." should be on top of the price, not next to it.

#### On Mobile - simulating screen size in Chrome is sufficient

1. Selected item should be bordered with blue and should have blue text for domain name.
2. Simulate selecting the recommended domain by adding "is-selected" class to the item.
4. All blue rule applies, but price should be on top of "included in paid plans". 
5. "First year included in paid plans" should be cut down to "included in paid plans".

### Testing Gutenboarding - Should be unaffected.

1. [Visit the Calypso live link](https://calypso.live/new?branch=update/domain-picker-ui-changes) of this PR.
2. In another tab, visit wordpress.com/new.
3. Compare domain picker. They should look exactly the same.

Partially fixes #46862

### CODE REVIEW

**Comment by ciampo** [comment]
> I'm not very happy with the term individual, I mean each element is on its own.
What about using `single` ?

**Comment by alshakero** [comment]
> What about using single ?

It's not necessary a single item. It's just not connected to others.

File: packages/domain-picker/src/domain-picker/suggestion-item.tsx [hunk 1]
```diff
@@ -65,6 +69,25 @@ const DomainPickerSuggestionItem: FunctionComponent< Props > = ( {
...
 
+	const freeDomainLabel =
+		type === ITEM_TYPE_INDIVIDUAL_ITEM
+			? __( 'Default', __i18n_text_domain__ )
+			: __( 'Free', __i18n_text_domain__ );
+
+	const firstYearIncludedInPaid = isMobile
+		? __( 'Included in paid plans', __i18n_text_domain__ )
+		: createInterpolateElement(
+				__( '<strong>First year included</strong> in paid plans', __i18n_text_domain__ ),
```
Comment (a8ci18n) [inline packages/domain-picker/src/domain-picker/suggestion-item.tsx:80]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'First year included in paid plans' )` *ES Score: 6*

File: packages/domain-picker/src/domain-picker/style.scss [hunk 2]
```diff
@@ -248,11 +286,31 @@
...
+		text-transform: unset;
+		// stylelint-disable-next-line scales/font-size
+		font-size: 12px; // typography-exception smaller size needed
+	}
+
+	&.is-selected {
+		.domain-picker__badge {
+			color: $accent-blue;
+			background-color: #d9e8f7;
+		}
```
Comment (ciampo) [inline packages/domain-picker/src/domain-picker/style.scss:312]: The contrast ratio of the badge in the "selected" state is not accessible

Screenshot 2020-11-05 at 19 55 55

**Review by ciampo — CHANGES_REQUESTED** [review]
Code LGTM I tested according to the instructions and everything went as expected.

Before merging, it needs to solve conflicts and to fix the contrast issue (see inline comment)

File: packages/domain-picker/src/domain-picker/style.scss (hunk 2 above)
Comment (ollierozdarz) [reply packages/domain-picker/src/domain-picker/style.scss:312]: Thanks for checking this @ciampo! Can we please try the foreground color of the label inside the badge (shown in the screenshot) as #0F6CB3. I believe that's what it is in the designs, although I could have copy/pasted it incorrectly (which I'm known to do!)

Comment (alshakero) [reply packages/domain-picker/src/domain-picker/style.scss:312]: Thanks, Ollie! You're right, I made a mistake assuming the color is the same of the border. I changed to what you recommend and it's better but contrast was still under the recommended threshold. I cheated a bit and changed the background from `d9e8f7` to `ddebf8`.

**Review by ciampo — APPROVED** [review]
Thank you for solving conflicts and addressing the contrast issue.

I'm wondering if these newly added shades of blue should be standardized / added to the calypso colour palette (I'll open a separate issue for that)

This PR LGTM and can be merged!

File: packages/domain-picker/src/domain-picker/style.scss [hunk 3]
```diff
@@ -248,11 +286,31 @@
...
+
+.domain-picker__suggestion-item.type-individual-item {
+	.domain-picker__badge {
+		color: var( --studio-green-80 );
+		background-color: var( --studio-green-5 );
+		// stylelint-disable-next-line scales/radii
+		border-radius: 4px;
+		text-transform: unset;
+		// stylelint-disable-next-line scales/font-size
+		font-size: 12px; // typography-exception smaller size needed
```
Comment (razvanpapadopol) [inline packages/domain-picker/src/domain-picker/style.scss:305]: `$font-body-extra-small` should be used instead

**Pull request merged**